dynamiqs/mesolve/rouchon.py

```diff
@@ -104,21 +114,32 @@ def backward_augmented(
         # phi: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
 
         # compute cached operators
-        # H, Hnh, M0, M0dag, M0rev: (b_H, 1, n, n)
+        # H, Hnh, M0, M0rev: (b_H, 1, n, n)
         H = self.H(t)
         Hnh = self.Hnh(H)
         M0 = self.M0(Hnh)
-        M0dag = self.M0dag(M0)
         M0rev = self.M0rev(Hnh)
 
+        # normalize the Kraus Map
+        if self.cholesky_normalization:
+            # Rrev, Trev: (b_H, 1, n, n)
+            Rrev = self.Rrev(M0rev)
+            Trev = self.Trev(Rrev)
+            rho = inv_mm_invmh(Trev.mH, rho)  # Trev.mH^-1 @ rho @ Trev^-1
```

Comment:
`Trev` is upper triangular, so `Trev.mH` is lower, and this doesn't work with your function `inv_mm_invmh`.